### PR TITLE
#597: Move workflows into ci-build.yml

### LINKED ISSUE #597: Integrate PR jobs into `ci-build.yml`
Currently each project using PK must configure branch protection to contain jobs `build`, `next-java-compatibility` and `linkChecker`. To simplify configuration, PK should inline jobs `next-java-compatibility` and `linkChecker` into `ci-build.yml` and make job `build` depend on these two.

This way, projects only need to add a single job (`build`) to branch protection.

PK must delete the not needed workflow `yml` files automatically.

PK must allow customizing the `next-java-compatibility` job via `.project-keeper.yml`. Some projects might require special command line options.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #597

### CODE REVIEW

File: project-keeper/src/main/resources/templates/.github/workflows/broken_links_checker.yml [hunk 1]
```diff
@@ -5,10 +5,6 @@ name: Broken Links Checker
 on:
   schedule:
     - cron: "0 5 * * 0"
-  push:
-    branches:
-      - main
-  pull_request:
```
[the remark this answers is not in the document]
Comment (ckunki) [reply project-keeper/src/main/resources/templates/.github/workflows/broken_links_checker.yml:11]: Can I still change the documentation of any project and verify if all links are valid before merging to main?

[the remark this answers is not in the document]
Comment (ckunki) [reply project-keeper/src/main/resources/templates/.github/workflows/broken_links_checker.yml:11]: Ah - I think [ci-build-db-version-matrix.yml](https://github.com/exasol/project-keeper/pull/605/files#diff-476c9b784650e858ef27fa457619bbf7018f1181ac1a56751f7184f78536e38eR122) is the answer.

File: project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml [hunk 2]
```diff
@@ -110,6 +110,78 @@ jobs:
           path: ${{ steps.build-pk-verify.outputs.release-artifacts }}
           retention-days: 5
 
+      - name: Configure link check
```
Comment (ckunki) [inline project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml:113]: Is there a way of avoiding the duplicate configuration of github-action-markdown-link-check?

File: project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml [hunk 3]
```diff
@@ -110,6 +110,78 @@ jobs:
...
+        with:
+          fetch-depth: 0
+      - name: Set up JDKs
+        id: setup-java
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: |
+            11
+            17
```
Comment (ckunki) [inline project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml:183]: Shouldn't these versions be defined by the entries in `.project-keeper.yml`?

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/files/JavaVersionExtractorTest.java [hunk 4]
```diff
@@ -16,8 +17,9 @@
 class JavaVersionExtractorTest {
 
     @Test
-    void noSources() {
-        assertThat(extract(), contains("17"));
+    void nonSource() {
```
Comment (ckunki) [inline project-keeper/src/test/java/com/exasol/projectkeeper/validators/files/JavaVersionExtractorTest.java:20]:
```suggestion
    void noSources() {
```

File: project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml (hunk 2 above)
Comment (kaklakariada) [reply project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml:113]: Yes, but I didn't want to do this in this PR.

File: project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml (hunk 3 above)
Comment (kaklakariada) [reply project-keeper/src/main/resources/templates/.github/workflows/ci-build.yml:183]: These versions are overwritten by PK based on property `java.version` in the `pom.xml`.

**Review by ckunki — APPROVED** [review]

**Pull request merged**